custom_components/ble_monitor/config_flow.py

```diff
@@ -0,0 +1,157 @@
+    CONF_HCI_INTERFACE,
+    CONF_BATT_ENTITIES,
+    CONF_REPORT_UNKNOWN,
+    CONF_RESTORE_STATE,
+    CONF_ENCRYPTION_KEY,
+    DOMAIN,
+    MAC_REGEX,
+    AES128KEY_REGEX,
+)
+
+DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_MAC): cv.matches_regex(MAC_REGEX),
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_ENCRYPTION_KEY): cv.matches_regex(AES128KEY_REGEX),
+        vol.Optional(CONF_TEMPERATURE_UNIT): cv.temperature_unit,
+    }
+)
+
+DOMAIN_SCHEMA = vol.Schema(
+    {
+        vol.Optional(
+            CONF_HCI_INTERFACE, default=DEFAULT_HCI_INTERFACE
+        ): cv.positive_int,
+        vol.Optional(CONF_DISCOVERY, default=DEFAULT_DISCOVERY): cv.boolean,
+        vol.Optional(CONF_ACTIVE_SCAN, default=DEFAULT_ACTIVE_SCAN): cv.boolean,
+        vol.Optional(
+            CONF_REPORT_UNKNOWN, default=DEFAULT_REPORT_UNKNOWN
+        ): cv.boolean,
+        vol.Optional(
+            CONF_BATT_ENTITIES, default=DEFAULT_BATT_ENTITIES
+        ): cv.boolean,
+        vol.Optional(CONF_ROUNDING, default=DEFAULT_ROUNDING): cv.boolean,
+        vol.Optional(CONF_DECIMALS, default=DEFAULT_DECIMALS): cv.positive_int,
+        vol.Optional(CONF_PERIOD, default=DEFAULT_PERIOD): cv.positive_int,
+        vol.Optional(CONF_LOG_SPIKES, default=DEFAULT_LOG_SPIKES): cv.boolean,
+        vol.Optional(CONF_USE_MEDIAN, default=DEFAULT_USE_MEDIAN): cv.boolean,
+        vol.Optional(CONF_RESTORE_STATE, default=DEFAULT_RESTORE_STATE): cv.boolean,
+        # vol.Optional(CONF_DEVICES, default=[]): vol.All(
+        #     cv.ensure_list, [DEVICE_SCHEMA]
+        # ),
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+@config_entries.HANDLERS.register(DOMAIN)
+class BLEMonitorConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return BLEMonitorOptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        _LOGGER.info("async_step_user")
+        errors = {}
+        if user_input is not None:
+            title = f"BLE Monitor"
+            await self.async_set_unique_id(title)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=title, data=user_input)
+
+        return self._show_user_form(errors)
+
+    async def async_step_import(self, user_input=None):
+        """Handle import."""
+        _LOGGER.info("async_step_import: %s", user_input)
+        return await self.async_step_user(user_input)
+
+    @callback
+    def _show_user_form(self, errors=None):
+        return self.async_show_form(
+            step_id="user", data_schema=DOMAIN_SCHEMA, errors=errors or {}
+        )
+
+class BLEMonitorOptionsFlow(config_entries.OptionsFlow):
+    """Handle SpeedTest options."""
```

Comment:
Hope you don't mind when I make suggestions during the development. 

Change to
"""Handle BLE Monitor options."""